From c21f57566f4a6fe8ae66d887c5757e492c3a1107 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 18 Feb 2011 03:36:47 +0000
Subject: [PATCH] sched: Avoid ABI change in 2.6.32.29

Hide changes in scheduler internals from genksyms.

Guard all the scheduler internals declared in <linux/sched.h> (struct
sched_group, struct sched_domain, struct sched_class, etc.) with
'#ifndef MODULE' to ensure modules really don't use them.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 include/linux/sched.h |    6 ++++++
 kernel/sched.c        |    4 ++++
 2 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/include/linux/sched.h b/include/linux/sched.h
index 4f96d18..508d511 100644
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -816,6 +816,8 @@ enum cpu_idle_type {
 	CPU_MAX_IDLE_TYPES
 };
 
+#ifndef MODULE /* modules must not use this */
+
 /*
  * sched-domains (multiprocessor balancing) declarations:
  */
@@ -897,7 +899,9 @@ struct sched_group {
 	 * single CPU.
 	 */
 	unsigned int cpu_power;
+#ifndef __GENKSYMS__
 	unsigned int group_weight;
+#endif
 
 	/*
 	 * The CPUs this group covers.
@@ -1161,6 +1165,8 @@ struct sched_class {
 };
 #endif /* __GENKSYMS__ */
 
+#endif /* !MODULE */
+
 struct load_weight {
 	unsigned long weight, inv_weight;
 };
diff --git a/kernel/sched.c b/kernel/sched.c
index ecc5ffc..c937a96 100644
--- a/kernel/sched.c
+++ b/kernel/sched.c
@@ -528,7 +528,9 @@ struct rq {
 	struct mm_struct *prev_mm;
 
 	u64 clock;
+#ifndef __GENKSYMS__
 	u64 clock_task;
+#endif
 
 	atomic_t nr_iowait;
 
@@ -536,7 +538,9 @@ struct rq {
 	struct root_domain *rd;
 	struct sched_domain *sd;
 
+#ifndef __GENKSYMS__
 	unsigned long cpu_power;
+#endif
 
 	unsigned char idle_at_tick;
 	/* For active balancing */
-- 
1.7.4.1

